src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -0,0 +1,451 @@
+                            DataType = inputBinding.Metadata.DataType ?? DataType.String,
+                            Value = stream
+                        };
+                        await inputBinding.BindAsync(bindingContext);
+                    }
+                }
+
+                environmentVariables[inputdata] = Path.Combine(functionInstanceOutputPath, inputdata);
+            }
+        }
+
+        private static async Task ProcessOutputBindingsAsync(string functionInstanceOutputPath, Collection<FunctionBinding> outputBindings,
+            object input, IBinderEx binder, Dictionary<string, string> bindingData)
+        {
+            if (outputBindings == null)
+            {
+                return;
+            }
+
+            try
+            {
+                foreach (var outputBinding in outputBindings)
+                {
+                    string filePath = System.IO.Path.Combine(functionInstanceOutputPath, outputBinding.Metadata.Name);
+                    if (File.Exists(filePath))
+                    {
+                        using (FileStream stream = File.OpenRead(filePath))
+                        {
+                            BindingContext bindingContext = new BindingContext
+                            {
+                                TriggerValue = input,
+                                Binder = binder,
+                                BindingData = bindingData,
+                                Value = stream
+                            };
+                            await outputBinding.BindAsync(bindingContext);
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                // clean up the output directory
+                if (outputBindings.Any() && Directory.Exists(functionInstanceOutputPath))
+                {
+                    Directory.Delete(functionInstanceOutputPath, recursive: true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Event handler for the output stream.
+        /// </summary>
+        private void OutputCollectionDataAdded(object sender, DataAddedEventArgs e, TraceWriter traceWriter)
+        {
+            // do something when an object is written to the output stream
+            var source = (PSDataCollection<PSObject>)sender;
+            var msg = source[e.Index].ToString();
+            traceWriter.Info(msg);
+            TraceWriter.Info(msg);
+        }
+
+        /// <summary>
+        /// Event handler for the error stream.
+        /// </summary>
+        private void ErrorDataAdded(object sender, DataAddedEventArgs e, TraceWriter traceWriter)
+        {
+            var source = (PSDataCollection<ErrorRecord>)sender;
+            var msg = GetErrorMessage(source[e.Index]);
+            traceWriter.Error(msg);
+            TraceWriter.Error(msg);
+        }
+
+        private string GetErrorMessage(ErrorRecord errorRecord)
```

Comment:
Need unit tests for all of these complicated formatting functions. Make them internal and add tests.